chain/xrp/tx/tx.go

```diff
@@ -0,0 +1,124 @@
+	Account     xc.Address          `json:"Account,omitempty"`     // Sending account address
+	Amount      xc.AmountBlockchain `json:"Amount,omitempty"`      // Amount to deliver
+	Destination xc.Address          `json:"Destination,omitempty"` // Destination account
+	Fee         string              `json:"Fee,omitempty"`         // Optional fee (if none, XRPL calculates it)
+	Flags       int                 `json:"Flags"`                 // Flags for this transaction
+	//LastLedgerSequence int                 `json:"LastLedgerSequence"`        // Optional last ledger sequence
+	Sequence        int             `json:"Sequence,omitempty"`        // Sequence number of the account
+	SigningPubKey   string          `json:"SigningPubKey,omitempty"`   // Public key of the account
+	TransactionType TransactionType `json:"TransactionType,omitempty"` // Type of the transaction
+	TxnSignature    string          `json:"TxnSignature,omitempty"`    // Transaction signature
+	//AccountTxnID       string              `json:"AccountTxnID"`       // Hash of a previous transaction
+	//DeliverMin     *Amount `json:"DeliverMin,omitempty"`     // Optional minimum amount to be delivered
+	//DestinationTag *uint32 `json:"DestinationTag,omitempty"` // Optional destination tag
+	//InvoiceID      *string `json:"InvoiceID,omitempty"`      // Optional invoice ID (256-bit hash)
+	//Memos              *[]Memo         `json:"memos,omitempty"`              // Optional memos
+	//NetworkID      *uint32 `json:"NetworkID,omitempty"`      // Optional network ID
+	//Paths   *[]Path `json:"Paths,omitempty"`   // Optional paths for cross-currency payments
+	//SendMax *Amount `json:"SendMax,omitempty"` // Optional maximum amount allowed to be spent
+	//Signers        *[]Signer `json:"signers,omitempty"`         // Optional signers for multisig
+	//SourceTag      *uint32 `json:"SourceTag,omitempty"`      // Optional source tag
+	//TicketSequence *uint32 `json:"TicketSequence,omitempty"` // Optional ticket sequence
+}
+
+// Tx for Template
+type Tx struct {
+	XRPTx                *XRPTransaction
+	SignPubKey           []byte
+	EncodeForSigning     []byte
+	EncodeXRPTx          *string
+	TransactionSignature []xc.TxSignature
+}
+
+var _ xc.Tx = &Tx{}
+
+// Hash returns the tx hash or id
+func (tx Tx) Hash() xc.TxHash {
+	return xc.TxHash("not implemented")
+}
+
+// Sighashes returns the tx payload to sign, aka sighash
+func (tx Tx) Sighashes() ([]xc.TxDataToSign, error) {
+	if tx.XRPTx == nil {
+		return nil, errors.New("missing XRP transaction")
+	}
+
+	if tx.EncodeForSigning == nil {
+		return nil, errors.New("missing serialised XRP transaction")
+	}
+
+	hash := sha256.Sum256(tx.EncodeForSigning)
+
+	return []xc.TxDataToSign{hash[:]}, nil
+
+}
+
+// AddSignatures adds a signature to Tx
+func (tx *Tx) AddSignatures(signatures ...xc.TxSignature) error {
+	for _, sig := range signatures {
+		signatureHex := hex.EncodeToString(sig)
+		tx.XRPTx.TxnSignature = signatureHex
```

Comment:
It looks like XRP uses DER encoding so we're need to convert it.  See example transaction, the binary byte length is not 64/65, but over 70 bytes.

https://xrpscan.com/tx/BE63DB29A62C16E02290562B28CFECA8870D65E8E1E0A605788643AAFE57C664

Bitcoin also uses DER so you can take example from that chain.

```go
	for i, rsvBytes := range signatures {
		r, s, err := DecodeEcdsaSignature(rsvBytes)
		if err != nil {
			return err
		}

		signature := ecdsa.NewSignature(&r, &s)
                tx.XRPTx.TxnSignature = signature.Serialize()
                // ...
    }
```

Reply:
Yes, you are right:
b'0D\x02 :\xc7\x0eC\x965\x0b\x1d\xb8l\x8b\xd1\x17~\xcf\x8fd6\xe1\x9e\xcf\xfa\xc5\xf3&M>\xae\x88|c\r\x02 )\xc1\x87x\x99\xa5\xab\xc6|=\xb4\xd3\x8f0\xa4\x98\x8d\x86\xcc\x96_O\x19\xf9i6\xad\xe6X\x85[\xa5'
.
This was obtained from Python library and it has 70bytes, length.